src/components/Scheme/ContributionRewardExtRewarders/Competition/utils.ts

```diff
@@ -0,0 +1,190 @@
+    if (now.isSameOrAfter(submissionsEndTime)) {
+      paused = true;
+      text = "Paused";
+    } else {
+      open = true;
+      text = "Open for suggestions";
+    }
+  } else if (now.isBefore(endTime)) {
+    voting = true;
+    text = "Voting started!";
+  } else {
+    complete = true;
+    text = "Complete";
+  }
+
+  return {
+    complete,
+    endTime,
+    now,
+    open,
+    paused,
+    submissionsEndTime,
+    startTime,
+    text,
+    voting,
+    votingStartTime,
+  };
+};
+
+export interface ICreateSubmissionOptions {
+  description: string;
+  title: string;
+  url: string;
+  tags: Array<string>;
+}
+
+export const createCompetitionSubmission = (proposalId: string, options: ICreateSubmissionOptions ): ThunkAction<any, IRootState, null> => {
+  return async (dispatch: Redux.Dispatch<any, any>, _getState: () => IRootState) => {
+    try {
+      const observer = operationNotifierObserver(dispatch, "Create Submission");
+      const competition = new Competition(proposalId, getArc());
+      await competition.createSuggestion(options).subscribe(...observer);
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error(err);
+      throw err;
+    }
+  };
+};
+
+export interface IVoteSubmissionOptions {
+  id: string; // actual id, not the counter
+}
+
+export const voteForSubmission = (options: IVoteSubmissionOptions ): ThunkAction<any, IRootState, null> => {
+  return async (dispatch: Redux.Dispatch<any, any>, _getState: () => IRootState) => {
+    try {
+      const observer = operationNotifierObserver(dispatch, "Vote Submission");
+      const submission = new CompetitionSuggestion(options.id, getArc());
+
+      await submission.vote().subscribe(...observer);
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error(err);
+      throw err;
+    }
+  };
+};
+
+export interface IVoteSubmissionOptions {
+  id: string; // actual id, not the counter
+}
+
+export const redeemForSubmission = (options: IVoteSubmissionOptions ): ThunkAction<any, IRootState, null> => {
+  return async (dispatch: Redux.Dispatch<any, any>, _getState: () => IRootState) => {
+    try {
+      const observer = operationNotifierObserver(dispatch, "Redeem Submission");
+      const submission = new CompetitionSuggestion(options.id, getArc());
+
+      await submission.redeem().subscribe(...observer);
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error(err);
+      throw err;
+    }
+  };
+};
+
+
+/**
+ * must be an exact subset of ICompetitionSuggestionQueryOptions
+ */
+export interface IGetSubmissionsOptions {
+  id?: string; // id of the competition
+  suggestionId?: string; // the "suggestionId" is a counter that is unique to the scheme
+  // - and is not to be confused with suggestion.id
+}
+
+const getSubmissions = (
+  proposalId: string,
+  options?: IGetSubmissionsOptions,
+  subscribe = false
+): Observable<Array<ICompetitionSuggestion>> => {
```

Comment:
It seems that with the definitions in this `utils` file you are mostly:
- fixing things that should have been handled in the client (which given the history of developing this stuff is understandable and good, but should not be considered the deifnitive solution)
-  wrapping the client functions (which is not a good idea, as it obfuscates the code)

Ideally, we'd not have the `getSubmissions`, `getProposalSubmissions` etc because there would be functios aviable from the client.